jwql/instrument_monitors/miri_monitors/cosmic_ray_monitor.py

```diff
@@ -0,0 +1,467 @@
+from sqlalchemy import func
+from sqlalchemy.sql.expression import and_
+
+# Functions for file manipulation
+from jwql.utils.utils import copy_files, ensure_dir_exists, get_config
+
+# Astropy functions
+from astropy.io import fits
+from astropy.time import Time
+
+# Utility functions
+from jwql.instrument_monitors import pipeline_tools
+import os
+import numpy as np
+import shutil
+import julian
+import datetime
+
+@log_fail
+@log_info
+def cosmic_ray_monitor():
+    """
+        The main function of the ``monitor_template`` module.
+        Queries MAST for new MIRI data and copies it to a working
+        directory where it is run through the JWST pipeline. The output of the 'jump'
+        and 'rate' steps is used to determine the number and magnitudes of cosmic rays
+        which is then saved to the database.
+        """
+    
+    logging.info('Begin logging for cosmic_ray_monitor')
+    
+    aperture = 'MIRIM_FULL'
+    query_table = MIRICosmicRayQueryHistory
+    stats_table = MIRICosmicRayStats
+    
+    #We start by querying MAST for new data
+    start_date = most_recent_search(aperture, query_table)
+    end_date = Time.now().mjd
+    logging.info('\tMost recent query: {}'.format(start_date))
+    
+    new_entries = query_mast(start_date, end_date)
+    
+    #for testing purposes only
+    new_filenames = get_config()['local_test_data']
+    
+    #for file_entry in new_entries['data']:
+    #    try:
+    #        new_filenames.append(filesystem_path(file_entry['filename']))
+    #    except FileNotFoundError:
+    #        logging.info('\t{} not found in target directory'.format(file_entry['filename']))
+    #    except ValueError:
+    #        logging.info(
+    #            '\tProvided file {} does not follow JWST naming conventions.'.format(file_entry['filename']))
+    
+    #Next we copy new files to the working directory
+    output_dir = os.path.join(get_config()['outputs'], 'cosmic_ray_monitor')
+    
+    data_dir = get_config()['local_test_dir'] #for testing purposes only
+    #data_dir =  os.path.join(output_dir,'data')
+    
+    ensure_dir_exists(data_dir)
+    cosmic_ray_files, not_copied = copy_files(new_filenames, data_dir)
+    
+    for file_name in cosmic_ray_files:
+        if 'uncal' in file_name:
+            dir_name = file_name[51:76] #Gets the observation identifier. Should be constant?
+            obs_dir = os.path.join(data_dir, dir_name)
+            ensure_dir_exists(obs_dir)
+            
+            head = fits.getheader(file_name)
+            nints = head['NINTS']
+            
+            try:
+                shutil.copy2(file_name, obs_dir)
+            except:
+                logging.info('Failed to copy {} to observation dir.'.format(file_name))
+        
+            #Next we run the pipeline on the files to get the proper outputs
+            MIRI_steps = pipeline_tools.get_pipeline_steps('MIRI')
+            
+            file = os.path.join(obs_dir, os.path.basename(file_name))
```

Comment:
I think `file` is a built-in python keyword and thus should be avoided being used as a variable name.  Could you use something different here?

Reply:
Changed to `uncal_file`